packages/realm/src/app-services/User.ts

```diff
@@ -58,7 +61,25 @@ export interface UserIdentity {
   providerType: ProviderType;
 }
 
+/** @internal */
+function cleanArguments(args: unknown[] | unknown): unknown[] | unknown {
+  if (Array.isArray(args)) {
+    return args.map((x) => cleanArguments(x));
+  } else if (typeof args === "object") {
+    const result: { [key: string]: unknown } = {};
+    for (const [k, v] of Object.entries(args as object)) {
+      if (typeof v !== "undefined") {
+        result[k] = v;
```

Comment:
Shouldn't this be `result[k] = cleanArguments(v)`? Or is there a reason we don't want to recursively strip undefineds?

Reply:
This method was extracted from `utils.js` which looks like the following:
<img width="300" alt="image" src="https://user-images.githubusercontent.com/44238293/201334418-d27b63fc-b3be-43d9-86dc-d3c612e7a158.png">
We actually don't call it there either, but your reasoning seems right and I think we should unless there's no reason to why we don't do it..